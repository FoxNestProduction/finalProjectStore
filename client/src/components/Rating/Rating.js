import React, { useState } from 'react';
import Box from '@mui/material/Box';
import Rating from '@mui/material/Rating';
import PropTypes from 'prop-types';
import StarIcon from '../../assets/svgComponents/StarIcon';

const RatingItem = ({ ratingValue }) => {
  const [value, setValue] = useState(ratingValue);

  return (
    <Box>
      <Rating
        sx={{
          width: {
            mobile: 140,
            tablet: 140,
            desktop: 150,
          },
          marginLeft: '-4px',
        }}
        value={value}
        onChange={(event, newValue) => {
          setValue(newValue);
        }}
        // precision={0.5}
        size="small"
        icon={<StarIcon fill="#6C5FBC" />}
        emptyIcon={<StarIcon fill="#e7ddc5" />}
        readOnly
      />
    </Box>
  );
};

RatingItem.propTypes = {
  ratingValue: PropTypes.number,
};

RatingItem.defaultProps = {
  ratingValue: 3,
};

export default RatingItem;
