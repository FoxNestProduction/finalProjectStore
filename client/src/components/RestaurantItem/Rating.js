import * as React from 'react';
import Box from '@mui/material/Box';
import Rating from '@mui/material/Rating';
import { PropTypes } from 'prop-types';
import StarIcon from '../SvgComponents/StarIcon';

const RatingItem = ({ ratingValue }) => {
  const [value, setValue] = React.useState(3);

  const changeRating = () => {
    setValue(ratingValue);
  };

  return (
    <Box>
      <Rating
        value={value}
        // value={changeRating()}
        onChange={(event, newValue) => {
          setValue(newValue);
        }}
        // precision={0.5}
        size="large"
        icon={<StarIcon fill="#6C5FBC" />}
        emptyIcon={<StarIcon fill="#e7ddc5" />}
        readOnly
      />
    </Box>
  );
};

RatingItem.propTypes = {
  ratingValue: PropTypes.number,
};

RatingItem.defaultProps = {
  ratingValue: null,
};

export default RatingItem;
