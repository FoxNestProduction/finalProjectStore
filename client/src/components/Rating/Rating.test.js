import React from 'react';
import { render } from '@testing-library/react';
import Rating from './Rating';

describe('smoke test', () => {
  test('should Rating render', () => {
    const { asFragment } = render(<Rating ratingValue="5" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
