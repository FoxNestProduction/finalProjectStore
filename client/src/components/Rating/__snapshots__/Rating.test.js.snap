// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke test should Rating render 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <span
      aria-label="5 Stars"
      class="MuiRating-root MuiRating-sizeSmall Mui-readOnly MuiRating-readOnly css-1nmdjm5-MuiRating-root"
      role="img"
    >
      <span>
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="StarRateRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
            />
          </svg>
        </span>
      </span>
      <span>
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="StarRateRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
            />
          </svg>
        </span>
      </span>
      <span>
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="StarRateRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
            />
          </svg>
        </span>
      </span>
      <span>
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="StarRateRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
            />
          </svg>
        </span>
      </span>
      <span>
        <span
          class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="StarRateRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
            />
          </svg>
        </span>
      </span>
    </span>
  </div>
</DocumentFragment>
`;
